### PR TITLE
feature/CVSB-8650 - As a support team user, I want to switch to a different technical record

### PR DESCRIPTION
**Pull request by irina005**
Update:
- rebased with develop
- agreed with Umer and Olu to merge this after we merge ADR as we have to sync the changes in these files

### CODE REVIEW

File: src/app/technical-record/tech-rec-history/tech-rec-history.component.spec.ts [hunk 1]
```diff
@@ -38,37 +42,52 @@ describe('VehicleSummaryComponent', () => {
...
     };
-
     fixture.detectChanges();
   });
 
   it('should create my component', async(() => {
     expect(component).toBeTruthy();
   }));
 
+  it('fakeAsync works', fakeAsync(() => {
```
Comment (alleypa) [inline src/app/technical-record/tech-rec-history/tech-rec-history.component.spec.ts]: I dont think you should test fakeAsync

File: src/app/technical-record/technical-record.component.ts [hunk 2]
```diff
@@ -69,6 +42,38 @@ export class TechnicalRecordComponent implements OnInit, OnDestroy, ComponentCan
   numberFee: any;
   isAdrNull: any;
   navigationSubscription;
+  queryParamsSubscription: any;
+  queryParamId = 0;
```
Comment (alleypa) [inline src/app/technical-record/technical-record.component.ts]: the value is the searchIdentifier right?

**Comment by irina005** [comment]
Screenshot 2020-03-02 at 13 51 55

File: src/app/technical-record/technical-record.component.ts [hunk 3]
```diff
@@ -1,116 +1,132 @@
...
+
+  constructor(
+    private dialog: MatDialog,
+    private allowedValues: TechnicalRecordValuesMapper,
+    public techRecHelpers: TechRecordHelpersService,
+    private route: ActivatedRoute
+  ) {}
+
+  ngOnInit() {
+    this.queryParamsSubscription  = this.route.params.subscribe(params => {
```
Comment (alleypa) [inline src/app/technical-record/technical-record.component.ts:68]: we should not use this solution post ADR merge.

File: src/app/technical-record/technical-record.component.ts [hunk 4]
```diff
@@ -1,116 +1,132 @@
...
+      this.queryParamId = isNaN(params['id']) ? 0 : +params['id'];
+    });
+    this.activeRecord = this.currentRecords[this.queryParamId];
```
Comment (alleypa) [inline src/app/technical-record/technical-record.component.ts:71]: Better to leave the existing code as is and adopt ngrx state management update.

File: src/app/app.component.scss [hunk 5]
```diff
@@ -485,6 +485,12 @@ a#cancel2 {
   display: block !important;
 }
 
+.tech-rec-status {
```
Comment (alleypa) [inline src/app/app.component.scss:488]: less code in the file.

File: src/app/technical-record/body/body.component.spec.ts [hunk 6]
```diff
@@ -1,26 +1,28 @@
...
-import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
-import { MaterialModule } from '@app/material.module';
-import { SharedModule } from '@app/shared/shared.module';
-import { RouterTestingModule } from '@angular/router/testing';
-import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
-import { NgrxFormsModule } from 'ngrx-forms';
-import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
-import { VehicleSummaryComponent } from '@app/technical-record/vehicle-summary/vehicle-summary.component';
-import { TechnicalRecordComponent } from '@app/technical-record/technical-record.component';
+import {async, ComponentFixture, getTestBed, TestBed} from '@angular/core/testing';
```
Comment (alleypa) [inline src/app/technical-record/body/body.component.spec.ts:1]: run prettier format on this file

File: src/app/technical-record/body/body.component.spec.ts [hunk 7]
```diff
@@ -1,26 +1,28 @@
...
+import {MatDialogModule} from '@angular/material/dialog';
+import {FormsModule, ReactiveFormsModule} from '@angular/forms';
+import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
+import {MaterialModule} from '@app/material.module';
+import {SharedModule} from '@app/shared/shared.module';
+import {RouterTestingModule} from '@angular/router/testing';
+import {FontAwesomeModule} from '@fortawesome/angular-fontawesome';
+import {NgrxFormsModule} from 'ngrx-forms';
+import {CUSTOM_ELEMENTS_SCHEMA} from '@angular/core';
+import {TechnicalRecordComponent} from '@app/technical-record/technical-record.component';
```
Comment (alleypa) [inline src/app/technical-record/body/body.component.spec.ts:15]: I think we should remove this TechnicalRecordComponent import  and also any non needed import

File: src/app/technical-record/dimensions/dimensions.component.spec.ts [hunk 8]
```diff
@@ -38,15 +38,15 @@ describe('VehicleSummaryComponent', () => {
         ReactiveFormsModule,
         NgrxFormsModule
       ],
-      declarations: [VehicleSummaryComponent, TechnicalRecordComponent],
+      declarations: [DimensionsComponent, TechnicalRecordComponent],
       providers: [
         TechRecordHelpersService
```
Comment (alleypa) [inline src/app/technical-record/dimensions/dimensions.component.spec.ts:43]: Better to stop using this TechRecordHelpersService

File: src/app/technical-record/dimensions/dimensions.component.spec.ts [hunk 9]
```diff
@@ -12,13 +12,13 @@ import {RouterTestingModule} from '@angular/router/testing';
 import {FontAwesomeModule} from '@fortawesome/angular-fontawesome';
 import {NgrxFormsModule} from 'ngrx-forms';
 import {CUSTOM_ELEMENTS_SCHEMA} from '@angular/core';
-import {VehicleSummaryComponent} from '@app/technical-record/vehicle-summary/vehicle-summary.component';
 import {TechnicalRecordComponent} from '@app/technical-record/technical-record.component';
```
Comment (alleypa) [inline src/app/technical-record/dimensions/dimensions.component.spec.ts:15]: TechnicalRecordComponent not needed as import in this spec file

File: src/app/technical-record/tech-rec-history/tech-rec-history.component.html [hunk 10]
```diff
@@ -29,7 +29,9 @@
       <td id="{{ 'test-createdAt-' + i }}" class="govuk-table__cell">
         {{ techRec.createdAt ? (techRec.createdAt | date: 'dd/MM/yyyy') : '-' }}
       </td>
-      <td class="govuk-table__cell"></td>
+      <td class="govuk-table__cell">
+        <a class="govuk-link" id="{{'tech-rec-'+i}}" (click)="switchTechRecord(i)" [hidden]="i==0">View</a>
```
Comment (alleypa) [inline src/app/technical-record/tech-rec-history/tech-rec-history.component.html:33]: use store/action approach

File: src/app/technical-record/tech-rec-history/tech-rec-history.component.spec.ts [hunk 11]
```diff
@@ -38,37 +42,42 @@ describe('VehicleSummaryComponent', () => {
...
-      'vehicleId': 'JY58FPP',
-      'vehicleType': 'psv',
-      'axles': [
-        { 'parkingBrakeMrk': false, 'axleNumber': 1 },
-        { 'parkingBrakeMrk': true, 'axleNumber': 2 },
-        { 'parkingBrakeMrk': false, 'axleNumber': 3 }
-      ]
-    };
-
+    component.techRecordsJson = [{
```
Comment (alleypa) [inline src/app/technical-record/tech-rec-history/tech-rec-history.component.spec.ts:58]: We should TESTING.Utils

File: src/app/technical-record/tech-rec-history/tech-rec-history.component.spec.ts [hunk 12]
```diff
@@ -38,37 +42,42 @@ describe('VehicleSummaryComponent', () => {
...
+      }]
+    }];
     fixture.detectChanges();
   });
 
   it('should create my component', async(() => {
     expect(component).toBeTruthy();
   }));
 
+  it('should call the action to switch to another technical record', fakeAsync(() => {
```
Comment (alleypa) [inline src/app/technical-record/tech-rec-history/tech-rec-history.component.spec.ts:74]: this wont be need for the store approach

File: src/app/technical-record/tech-record-helpers.service.ts [hunk 13]
```diff
@@ -19,4 +19,9 @@ export class TechRecordHelpersService {
   capitalizeWord(word: string) {
     return word.charAt(0).toUpperCase() + word.substr(1).toLowerCase();
   }
+
+  scrollDown() {
```
Comment (alleypa) [inline src/app/technical-record/tech-record-helpers.service.ts:23]: we should move this to tech record history file

File: src/app/technical-record/technical-record.component.spec.ts [hunk 14]
```diff
@@ -13,6 +13,7 @@ import { Store, StoreModule, INITIAL_STATE } from '@ngrx/store';
 import { hot } from 'jasmine-marbles';
 import { of, Subject } from 'rxjs';
 import { TechnicalRecordComponent } from './technical-record.component';
+import { TechRecordHelpersService } from '@app/technical-record/tech-record-helpers.service';
```
Comment (alleypa) [inline src/app/technical-record/technical-record.component.spec.ts:16]: not needed

File: src/app/technical-record/technical-record.container.spec.ts [hunk 15]
```diff
@@ -135,14 +135,15 @@ describe('TechnicalRecordsContainer', () => {
   selector: 'vtm-technical-record',
   template: `
     <div>Current or Selected Vehicle Tech Record: {{ vehicleTechRecord | json }}</div>
-    <div>Active record based on Statuscode: {{ activeRecord | json }}</div>
+<!--    <div>Active record based on Statuscode: {{ activeRecord | json }}</div>-->
```
Comment (alleypa) [inline src/app/technical-record/technical-record.container.spec.ts:138]: undo

**Comment by alleypa** [comment]
closing in preference for CVSB-16683

**Pull request closed without merging**